### PR TITLE
Git inter team tests

### PR DESCRIPTION
**Pull request by dennari**
Adds tests which ensure users can only push to / clone repositories they are authorized to.

### CODE REVIEW

File: src/authentication/git-auth-scheme.ts [hunk 1]
```diff
@@ -128,7 +139,7 @@ export class GitAuthScheme {
       body: JSON.stringify(body),
     });
     const json = await getResponseJson<{ access_token: string }>(response);
-    return { accessToken: json.access_token as string };
+    return json.access_token as string;
```
Comment (vsaarinen) [inline src/authentication/git-auth-scheme.ts]: Isn't the `as string` unnecessary?

File: src/integration-test/configuration.staging.ts_ [hunk 2]
```diff
@@ -0,0 +1,47 @@
+export default 's3://lucify-configuration/charles/configuration.staging.json';
```
Comment (vsaarinen) [inline src/integration-test/configuration.staging.ts_]: Is this supposed to be here?

File: src/integration-test/tests-inter-team.ts [hunk 3]
```diff
@@ -35,6 +36,95 @@ export default (
...
+        await runCommand('git', '-C', repoFolder, 'remote', 'rm', 'origin');
+        await runCommand('touch', `${repoFolder}/newFile${Date.now()}`);
+        await runCommand('git', '-C', repoFolder, 'add', '-A');
+        await runCommand('git', '-C', repoFolder, 'commit', '-m', 'message');
+        try {
+          await runCommand('git', '-C', repoFolder, 'push', 'minard', 'master');
+          expect.fail(undefined, undefined, 'message');
+        } catch (error) {
+          expect(error.message).to.not.eq('message');
+          console.log(error);
```
Comment (vsaarinen) [inline src/integration-test/tests-inter-team.ts]: Is this supposed to be here?

File: src/integration-test/tests-inter-team.ts [hunk 4]
```diff
@@ -35,6 +36,95 @@ export default (
...
+        const repoUrl = clients.admin.getRepoUrlWithCredentials({
+          username: accessToken,
+          password: '',
+        });
+        try {
+          await runCommand('git', 'clone', repoUrl, repoFolder);
+          expect.fail(undefined, undefined, 'message');
+        } catch (error) {
+          expect(error.message).to.not.eq('message');
+          console.log(error);
```
Comment (vsaarinen) [inline src/integration-test/tests-inter-team.ts]: Is this supposed to be here?

File: src/integration-test/configuration.staging.ts_ (hunk 2 above)
Comment (dennari) [inline src/integration-test/configuration.staging.ts_]: God damn..

**Comment by juhoojala** [comment]
:+1:

Merging, but added one issue that can be fixed later: https://github.com/lucified/minard-backend/issues/224.

**Pull request merged**